src/PrestaShopBundle/Bridge/AdminController/AdminControllerTrait.php

```diff
@@ -97,45 +65,68 @@ public function addAction(ActionInterface $action): void
     }
 
     /**
-     * This method add action specific for the list.
+     * @param Request $request
      *
-     * @param ActionInterface $action
-     * @param HelperListConfiguration $helperListConfiguration
-     *
-     * @return void
+     * @return LegacyControllerBridgeInterface
      */
-    public function addActionList(ActionInterface $action, HelperListConfiguration $helperListConfiguration): void
+    public function initLegacyControllerBridge(Request $request): LegacyControllerBridgeInterface
```

Comment:
I don't think we should require the request on this method, the controller is supposed to know its own associated legacy controller, so it can provide this information without relying on a request attribute This would simply this public method

Reply:
```suggestion
    public function getControllerBridge(): LegacyControllerBridgeInterface
```

Instead I would recommend a simple getter method, that can be called by any other components that may need to access the bridge controller (configurations, other listerns, the Symfony controller itself). Of course to avoid building a different one each time you should rely on the private field to create and store it.

Also the trait can't know all the required parameters, so the public method should be on the interface but not implemented, the trait itself provides a protected method that helps build the bridge controller, which would look like this:

```php
    protected function buildControllerBridge(
        string $legacyControllerName,
        string $tableName,
        string $className
    ): LegacyControllerBridgeInterface {
        if (!$this->legacyControllerBridge) {
            // Build the bridge and set the $this->legacyControllerBridge
            ....
        }

        return $this->legacyControllerBridge;
    }
```

Maybe this method will evolve with more parameters as we discover all the requirements.

Reply:
This way the Symfony controller only has to implement the public method:
```php
    public function getControllerBridge(): LegacyControllerBridgeInterface
    {
        return $this->buildControllerBridge('AdminFeatures', 'ps_features', Feature::class);
    }
```

Reply:
The same principle is implementable for the controller configuration actually since it's the one that carries all the config and can be heavily reused in the actually controller actions:

```php
    public function getControllerConfiguration(): ControllerConfiguration
    {
        return $this->buildControllerConfiguration('AdminFeatures', 'ps_features', Feature::class);
    }

    // And the build method for the bridge controller could probably look like this then
    public function getControllerBridge(): LegacyControllerBridgeInterface
    {
        return $this->buildControllerBridge($this->getControllerConfiguration());
    }
```

With such implementation we don't need the `getTable` and `getClassName` anymore, only two methods: `getControllerConfiguration` and `getControllerBridge` which seem to reflect more the actual responsibility of such a controller.

Reply:
>I don't think we should require the request on this method, the controller is supposed to know its own associated legacy controller

hm right, but we still need to provide _legacy_controller in route as we use it for security annotations, so isn't that abit redundant to also require additional method to be implemented for it?

Reply:
You're right, except for annotations it gives us a more simple way to uniformize the annotations everywhere And permission annotations are already linked to the request so it doesn't induce a dependency that wasn't already here

But in this case, if we can build the configuration without depending on a request I believe it will block us less, most of the time when I see a dependency on a request I'm not comfortable because usually it's a mess after when you are in a context without a request